ros2bag/ros2bag/api/__init__.py

```diff
@@ -44,6 +44,10 @@ def dict_to_duration(time_dict: Optional[Dict[str, int]]) -> Duration:
     """Convert a QoS duration profile from YAML into an rclpy Duration."""
     if time_dict:
         try:
+            if time_dict['sec'] < 0:
+                raise ValueError('`sec` may not be a negative value.')
+            if time_dict['nsec'] < 0:
```

Comment:
If `sec` is zero couldn't `nsec` be zero? I think you'll want to use all possible time inputs and convert to a duration.

Reply:
Do you mean that 0 seconds and 0 nsecs is not a valid input?

Reply:
Sorry let me rephrase and dismiss: I was thinking that it could be possible to have a negative `nsec` but not sure why someone would enter the value this way, because you're pulling the values directly from the user input. My mistake.